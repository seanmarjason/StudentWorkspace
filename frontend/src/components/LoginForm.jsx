import {useState} from 'react';
import axios from 'axios';
import './LoginForm.css';

// Generate JSX code for error message
const ErrorMessage = (name, errors) => {
  name === errors.name && (
    <div className="error">{errors.message}</div>
  );
}

// Login form handler
const handleLoginSubmit = (event, setIsLoginSuccess, setGroupId) => {
  event.preventDefault();
  const url = 'http://localhost:3000/users/login';
  var { username, password } = document.forms[0];
  const formData = new FormData();
  
  formData.append('username', username.value);
  formData.append('password', password.value);

  const config = {
    headers: {
      'content-type': 'multipart/form-data'
    }
  };

  axios.post(url, formData, config)
    .then((response) => {
      if (response.status === 200 && response.data.status === 'Login successful!') {
        console.log('Login successful! Group ID is:', response.data.group_id);
        setIsLoginSuccess(true);
        setGroupId(response.data.group_id);
      }
    })
    .catch((error) => {
      console.log(error.response.status);
    });
}


const LoginForm = ({ setIsLoginSuccess, setGroupId }) => {
  // eslint-disable-next-line no-unused-vars
  const [errorMessages, setErrorMessages] = useState({});

  return (
    <div className="login-form form">
      <form onSubmit={(event) => handleLoginSubmit(event, setIsLoginSuccess, setGroupId)}>
        <div className="title">Login</div>
        <div className="input-container">
          <label>Username </label>
          <input type="text" name="username" required />
          <ErrorMessage name="username" errors={errorMessages} />
        </div>
        <div className="input-container">
          <label>Password </label>
          <input type="password" name="password" required />
          <ErrorMessage name="password" errors={errorMessages} />
        </div>
        <div className="button-container">
          <button type="submit">Login</button>
        </div>
      </form>
    </div>
  );
}

export { LoginForm };
