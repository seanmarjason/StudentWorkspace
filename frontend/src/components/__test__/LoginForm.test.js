import {render } from '@testing-library/react';
import { LoginForm } from '../LoginForm';

describe('Login Form' , () => {
  it('should match snapshot', async () => {
    const view = render(<LoginForm />)

    expect(view.asFragment()).toMatchSnapshot();
  });
  
})
  