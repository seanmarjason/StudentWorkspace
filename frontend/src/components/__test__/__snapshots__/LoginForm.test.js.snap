// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="login-form form"
  >
    <form>
      <div
        class="title"
      >
        Login
      </div>
      <div
        class="input-container"
      >
        <label>
          Username 
        </label>
        <input
          name="username"
          required=""
          type="text"
        />
      </div>
      <div
        class="input-container"
      >
        <label>
          Password 
        </label>
        <input
          name="password"
          required=""
          type="password"
        />
      </div>
      <div
        class="button-container"
      >
        <button
          type="submit"
        >
          Login
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
